### PR TITLE
Fix connecting to replay server & fix "Welcome to RuneScape" detection

### PR DESCRIPTION
**Pull request by Hubcapp**
There was a problem with connecting to the replay server if a World was defined with a nonstandard port. The original code did not expect the port to change unless it told it to, causing the client to be unable to connect and hang.

---

Also, I have moved detection of being "all the way logged in" from the message hook that looks for the string "Welcome to RuneScape" to a place that I had always intended for "player is all the way logged in" to be detected: off of opcode 182 SHOW_WELCOME_SCREEN.

This is superior for 3 reasons:
1. I was always annoyed that "Welcome to RuneScape" was no longer the first message displayed to the client. This commit resolves a 4 year old TODO to figure out a way to put custom logic UNDER that message.
2. "Welcome to RuneScape" is an unsuitable detection for if the player has recently logged in. The text also appears when undergoing a Make-over-Mage transformation.
3. On some private servers, the text "Welcome to RuneScape" does not appear at all.

182 SHOW_WELCOME_SCREEN always appears exactly one time and only at login. It happens slightly after the "Welcome to RuneScape" message, which is ideal.

### CODE REVIEW

File: src/Game/Replay.java [hunk 1]
```diff
@@ -1153,6 +1153,16 @@ public static void dumpRawOutputStream(byte[] b, int off, int len) {
         output_checksum.update(buffer.array());
         output.write(buffer.array());
         output.flush();
+
+        /*
+         // Debug viewing entire output stream
+        System.out.print("Writing Output Stream: ");
```
Comment (lmsv-mx123) [inline src/Game/Replay.java:1159]: Do we want to keep this debug code?

File: src/Game/Replay.java [hunk 2]
```diff
@@ -1163,6 +1173,16 @@ public static void dumpRawOutputStream(byte[] b, int off, int len) {
       output_checksum.update(buffer.array());
       output.write(buffer.array());
       output.flush();
+
+      /*
+      // Debug viewing entire output stream
+      System.out.print("Writing Output Stream: ");
```
Comment (lmsv-mx123) [inline src/Game/Replay.java:1179]: idem here, is it intended to have this for debug purposes or something?

**Review by lmsv-mx123 — COMMENTED** [review]
Overall looks quite good, great changes

File: src/Game/Replay.java (hunk 2 above)
Comment (Hubcapp) [reply src/Game/Replay.java:1179]: Yea, I actually specifically put it in in its own commit. I have had to uncomment it somewhat frequencly while debugging ISAAC sync.

Comment (Hubcapp) [reply src/Game/Replay.java:1179]: I'm figuring I may have to come back to it some day, or might be useful to uncomment.

Comment (lmsv-mx123) [reply src/Game/Replay.java:1179]: alright then

File: src/Game/Replay.java (hunk 1 above)
Comment (lmsv-mx123) [reply src/Game/Replay.java:1159]: conversation on other comment

**Review by lmsv-mx123 — APPROVED** [review]

**Pull request merged**